webknossos-datastore/app/com/scalableminds/webknossos/datastore/dataformats/Cube.scala

```diff
@@ -31,20 +31,23 @@ trait Cube {
     // Check if we are the last one to use this cube, if that is the case and the cube needs to be removed -> remove it
     this.synchronized {
       accessCounter -= 1
-      if (accessCounter == 0 && scheduledForRemoval) {
-        onFinalize()
```

Comment:
@fm3 The biggest issue was probably, that here `isRemoved = true` was missing^^

Adding this, I decided to move this block into `tryRemoval`.